release: bump to version 1.2.0rc1 (#6175)
python-poetry/poetry: CHANGELOG.md

# Change Log

## [1.2.0rc1] - 2022-08-16

### Added

> - Virtual environments can now be created even with empty project name ([#5856](https://github.com/python-poetry/poetry/pull/5856))

Reviewer: I think this change belongs to `Fixed` (or `Changed`) rather than `Added`?

Author: I wasn't sure about it. IMO an empty project name isn't useful and therefor it was never considered as supported. Now it is, that's why I put it into this section. But if there are strong feelings against this, I can change it.

Reviewer: No strong feelings, I'm fine with keeping it here.

- Added support for subdirectories in git dependencies ([#5172](https://github.com/python-poetry/poetry/pull/5172))

> - Added support for nushell on `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))

Reviewer: ```suggestion
- Added support for `nushell` on `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))
```

### Changed

> - extras and extras dependencies are now sorted in lockfile ([#6169](https://github.com/python-poetry/poetry/pull/6169))

Reviewer: ```suggestion
- Extras and extras dependencies are now sorted in lock file ([#6169](https://github.com/python-poetry/poetry/pull/6169))
```

- Removed code for the `export` command, which is now provided via plugin ([#6128](https://github.com/python-poetry/poetry/pull/6128))

> - Replaced Poetry's helper method canonicalize_name() by packaging.utils.canonicalize_name() ([#6022](https://github.com/python-poetry/poetry/pull/6022))

Reviewer: ```suggestion
- Replaced Poetry's helper method `canonicalize_name()` by `packaging.utils.canonicalize_name()` ([#6022](https://github.com/python-poetry/poetry/pull/6022))
```

### Fixed

> - tomlkit versions with memory leak are now avoided ([#6160](https://github.com/python-poetry/poetry/pull/6160))

Reviewer: ```suggestion
- `tomlkit` versions with memory leak are now avoided ([#6160](https://github.com/python-poetry/poetry/pull/6160))
```

- Fixed an issue where Windows drive mappings break virtual environment names ([#6110](https://github.com/python-poetry/poetry/pull/6110))
- Fixed an issue where symlinks in lock file were not resolved ([#5850](https://github.com/python-poetry/poetry/pull/5850))
- Fixed an issue where `poetry lock` fails without output ([#6058](https://github.com/python-poetry/poetry/pull/6058))
- Fixed an issue where neither Python nor created venv can be found, when using Python from MS Store ([#5931](https://github.com/python-poetry/poetry/pull/5931))

> - Improved error message of `poetry publish` in the event of an upload error. ([#6043](https://github.com/python-poetry/poetry/pull/6043))

Reviewer: ```suggestion
- Improved error message of `poetry publish` in the event of an upload error ([#6043](https://github.com/python-poetry/poetry/pull/6043))
```

> - Fixed a tomlkit regression resulting in inconsistent line endings ([#5870](https://github.com/python-poetry/poetry/pull/5870))

Reviewer: ```suggestion
- Fixed a `tomlkit` regression resulting in inconsistent line endings ([#5870](https://github.com/python-poetry/poetry/pull/5870))
```

### Docs

> - Documented the use of the "subdirectory" parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))

Reviewer: ```suggestion
- Documented the use of the `subdirectory` parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))
```

> - Documented tox config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))

Reviewer: ```suggestion
- Documented `tox` config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))
```

# Change Log

## [1.2.0rc1] - 2022-08-22

### Added

- Added support for subdirectories in git dependencies ([#5172](https://github.com/python-poetry/poetry/pull/5172))
- Added support for yanked releases and files (PEP-592) ([#5841](https://github.com/python-poetry/poetry/pull/5841))

> - Virtual environments can now be created even with empty project name ([#5856](https://github.com/python-poetry/poetry/pull/5856))

Reviewer 2: ```suggestion
- Virtual environments can now be created even with empty project names ([#5856](https://github.com/python-poetry/poetry/pull/5856))
```

> - Added support for `nushell` on `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))

Reviewer 2: ```suggestion
- Added support for `nushell` in `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))
```

### Changed

> - Poetry now fallback to gather metadata for dependencies via pep517 if parsing `pyproject.toml` fail ([#5834](https://github.com/python-poetry/poetry/pull/5834))

Reviewer 2: ```suggestion
- Poetry now falls back to gather metadata for dependencies via pep517 if parsing `pyproject.toml` fails ([#5834](https://github.com/python-poetry/poetry/pull/5834))
```

> - Replaced Poetry's helper method `canonicalize_name()` by `packaging.utils.canonicalize_name()` ([#6022](https://github.com/python-poetry/poetry/pull/6022))

Reviewer 2: ```suggestion
- Replaced Poetry's helper method `canonicalize_name()` with `packaging.utils.canonicalize_name()` ([#6022](https://github.com/python-poetry/poetry/pull/6022))
```

- Removed code for the `export` command, which is now provided via plugin ([#6128](https://github.com/python-poetry/poetry/pull/6128))

> - Extras and extras dependencies are now sorted in lock file ([#6169](https://github.com/python-poetry/poetry/pull/6169))

Reviewer 2: ```suggestion
- Extras and extras dependencies are now sorted in the lock file ([#6169](https://github.com/python-poetry/poetry/pull/6169))
```

> - Removed deprecated CLI options ([#6210](https://github.com/python-poetry/poetry/pull/6210))

Reviewer 2: ```suggestion
- Removed deprecated (1.2-only) CLI options ([#6210](https://github.com/python-poetry/poetry/pull/6210))
```

### Fixed

> - Fixed an issue where symlinks in lock file were not resolved ([#5850](https://github.com/python-poetry/poetry/pull/5850))

Reviewer 2: ```suggestion
- Fixed an issue where symlinks in the lock file were not resolved ([#5850](https://github.com/python-poetry/poetry/pull/5850))
```

- Fixed a `tomlkit` regression resulting in inconsistent line endings ([#5870](https://github.com/python-poetry/poetry/pull/5870))

> - Fixed an issue where `POETRY_PYPI_TOKEN_PYPI` environment variable wasn't picked up ([#5911](https://github.com/python-poetry/poetry/pull/5911))

Reviewer 2: ```suggestion
- Fixed an issue where the `POETRY_PYPI_TOKEN_PYPI` environment variable wasn't respected ([#5911](https://github.com/python-poetry/poetry/pull/5911))
```

> - Fixed an issue where neither Python nor created venv can be found, when using Python from MS Store ([#5931](https://github.com/python-poetry/poetry/pull/5931))

Reviewer 2: ```suggestion
- Fixed an issue where neither Python nor a managed venv can be found, when using Python from MS Store ([#5931](https://github.com/python-poetry/poetry/pull/5931))
```

- Improved error message of `poetry publish` in the event of an upload error ([#6043](https://github.com/python-poetry/poetry/pull/6043))
- Fixed an issue where `poetry lock` fails without output ([#6058](https://github.com/python-poetry/poetry/pull/6058))
- Fixed an issue where Windows drive mappings break virtual environment names ([#6110](https://github.com/python-poetry/poetry/pull/6110))
- `tomlkit` versions with memory leak are now avoided ([#6160](https://github.com/python-poetry/poetry/pull/6160))
- Fixed an infinite loop in the solver ([#6178](https://github.com/python-poetry/poetry/pull/6178))
- Fixed an issue where latest version was used instead of locked one for vcs dependencies with extras ([#6185](https://github.com/python-poetry/poetry/pull/6185))

### Docs

> - Documented the use of the `subdirectory` parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))

Reviewer 2: ```suggestion
- Document use of the `subdirectory` parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))
```

> - Documented `tox` config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))

Reviewer 2: ```suggestion
- Document suggested `tox` config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))
```
